File: src/main/kotlin/basic_control_structures/exercise_8.kt
package basic_control_structures

// Create an algorithm that reads a distance in meters and shows the corresponding values in other measures.
fun meterToOtherMeasures() {
    print("Distance in meters: ")
    val distance = readln().toDouble()

    val km: Double = distance / 1000
    val hm: Double = distance / 100
    val dam: Double = distance / 10
    val dm: Double = distance * 10
    val cm: Double = distance * 100
    val mm: Double = distance * 1000

    println("The distance of $distance corresponds to: ")
    println("$km km")
    println("$hm hm")
    println("$dam dam")
    println("$dm dm")
    println("$cm cm")
    println("$mm mm")
}

File: Dart/a_basic_control_structure/exercise_8.dart
// Create an algorithm that reads a distance in meters and shows the corresponding values in other measures.

import 'dart:io';

void main() {
  stdout.write('Distance in meters: ');
  double distance = double.parse(stdin.readLineSync()!);

  Map<String, double> units = {
    'km': 0.001,
    'hm': 0.01,
    'dam': 0.1,
    'dm': 10,
    'cm': 100,
    'mm': 1000
  };

  print('The distance of $distance corresponds to: ');
  units.forEach((unit, factor) {
    print('${convert(distance, factor)}$unit');
  });
}

double convert(double distance, double factor) {
  return distance * factor;
}
